specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,318 @@
+  /// The Name is the exception's class name.
+  /// In the JSON it is `exceptionDetail.exception.className`.
+  /// This is the empty string if the exception doesn't have a class name.
+  /// This can happen if the script throws a non-Error object such as `throw "abc";`
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+  /// The Message is the exception's message and potentially stack.
+  /// In the JSON it is exceptionDetail.exception.description.
+  /// This is the empty string if the exception doesn't have a description.
+  /// This can happen if the script throws a non-Error object such as throw "abc";.
+  [propget] HRESULT Message([out, retval] LPWSTR* value);
+
+  /// This will return all details of the exception as a JSON string.
+  /// In the case that script has thrown a non-Error object such as `throw "abc";`
+  /// or any other non-Error object, you can get object specific properties.
+  [propget] HRESULT ExceptionAsJSON([out, retval] LPWSTR* value);
+}
+
+/// This is the result for ExecuteScriptWithResult.
+[uuid(D2C59C5C-AD36-4CF4-87CF-2F5359F6D4CB), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptResult : IUnknown {
+
+  /// This property is true if ExecuteScriptWithResult successfully executed script with
+  /// no unhandled exceptions and the result is available in the ResultAsJson property
+  /// or via the TryGetResultAsString method.
+  /// If it is false then the script execution had an unhandled exception which you
+  /// can get via the Exception property.
+  [propget] HRESULT IsSuccess([out, retval] BOOL* value); 
+
+  /// If IsSuccess is true, then this property is the JSON representation of the result of the script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT ResultAsJson([out, retval] LPWSTR* jsonResult);
+
+  /// If IsSuccess is true and the result of script execution is a string, this method provides the value of the string result
+  /// and otherwise returns E_INVALIDARG.
+  HRESULT TryGetResultAsString([out, retval] LPWSTR* stringResult);
+
+  /// If IsSuccess is false, you can use this property to get the unhandled exception thrown by script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT Exception(
+      [out, retval] ICoreWebView2ExecuteScriptException** exception);
+}
+
+/// This is the callback for ExecuteScriptWithResult
+[uuid(CECDD25B-E6E8-4A4E-B890-BBF95932564F), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptWithResultCompletedHandler : IUnknown {
+
+  /// Provides the result of ExecuteScriptWithResult
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
+      [in] ICoreWebView2ExecuteScriptResult* result);
+}
+
+/// This is the interface for getting string and exception with ExecuteScriptWithResult
+[uuid(67E0B57B-1AC7-4395-9793-5E4EF9C4B7D9), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+  
+  /// Run JavaScript code from the javascript parameter in the current
```

Comment:
Just a bit of rewording. It looks like I need to update the existing ExecuteScriptAsync documentation as well.

  /// Run JavaScript code from the `javaScript` parameter in the current
  /// top-level document rendered in the WebView.  
  /// The result of the execution is returned asynchronously in the CoreWebView2ExecuteScriptResult object
  /// which has methods and properties to obtain the successful result of script execution as well as any
  /// unhandled JavaScript exceptions.
  /// If this method is
  /// run after the `NavigationStarting` event during a navigation, the script
  /// runs in the new document when loading it, around the time
  /// `ContentLoading` is run.  This operation executes the script even if
  /// `ICoreWebView2Settings::IsScriptEnabled` is set to `FALSE`.